### PR TITLE
Deleting unused classes

### PR DESCRIPTION
**Pull request by fsamuel-bs**
**Goals (and why)**: There are some classes that are not used anywhere in the codebase. We should delete them.

**Implementation Description (bullets)**: -

**Concerns (what feedback would you like?)**: I've checked that these classes are not used anywhere else in our codebase, but I might have missed somewhere.

**Where should we start reviewing?**: -

**Priority (whenever / two weeks / yesterday)**: Whenever.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1933)

### CODE REVIEW

**Comment by clockfort** [comment]
These are all used in the project we pulled Atlas out of. @berler do you have bandwidth to pull these classes into PTMetropolisRdbms and friends anytime soon, or should @fsamuel-bs abandon this commit

**Comment by fsamuel-bs** [comment]
There's a pull request to move these files internally. I've assigned both @clockfort and @berler to review it.

File: timestamp-impl/src/main/java/com/palantir/timestamp/RateLimitedTimestampService.java [hunk 1]
```diff
@@ -1,264 +0,0 @@
...
-import com.palantir.common.proxy.TimingProxy;
-import com.palantir.util.jmx.OperationTimer;
-import com.palantir.util.timer.LoggingOperationTimer;
-
-/**
- * This uses smart batching to queue up requests and send them all as one larger batch.
- * @author carrino
- */
-@ThreadSafe
-public class RateLimitedTimestampService implements TimestampService {
```
Comment (jeremyk-91) [inline timestamp-impl/src/main/java/com/palantir/timestamp/RateLimitedTimestampService.java]: I'd like to request to keep this in the OSS codebase for now, if we're proceeding with this PR in general. We might need/want this optimisation for timelock to be scalable

**Comment by hsaraogi** [comment]
@clockfort, @berler could you take a look to ensure that this does not cause any issues for the large internal product?

**Comment by fsamuel-bs** [comment]
This should be reviewable again. After this merges, I'll make the needed changes to our large internal consumer.

**Comment by fsamuel-bs** [comment]
I've set on deleting only `FutureClosableIteratorTask`, `ClosableMergedIterator` and `ThrowingKeyValueService`. The `KvTableMappingService` will be used in the project to remap table names in the future.

**Comment by jeremyk-91** [comment]
@gmaretic do you have a use for ThrowingKVS in relation to the recent startup ordering work?

Happy to lose the iterator task and iterators.

**Pull request merged**